### DISCUSSION TITLE
Secrets not being masked

### DISCUSSION THREAD

**Discussion by feniix** [concourse/concourse#6473] [Help & Support] [answered]
Hi all,

we have this concourse resource we created to interact with an internal api

```go
fmt.Fprintf(os.Stderr, "incoming request: %+v\n", req)
```

here the `req` contains the user and password passed to the api, which are pipeline vars from secrets that in theory should be masked, but they happen to not get masked 

I removed that line for now, but I am curious why the secrets may not get masked

This is how the output looks like when it is not masked (and I did not remove the line above) the XXXXXX was added by me

```
stderr:

incoming request: {Source:{OktaAuthUrl:XXXXXXXX OktaMachineUserId:XXXXXXXX OktaMachineUserPassword:XXXXXXXXX OktaClientId:XXXXXXXX OktaClientSecret:XXXXXXXXX DepcomUrl:https://someurl DepcomEnv:PROD DepcomTarget:transform-test-python} Version:<nil>}

Unable to call depcom-service: Get "https://someurl/v1/concourse/latest-published-release-summary?environmentName=PROD&serviceName=transform-test-python": dial tcp 240.100.100.100:443: i/o timeout
```

Any idea?

Thanks in advance

**Comment by evanchaoli** [comment]
Where does `req` come from? Concourse only redacts secrets that are fetched from credential managers and `var_sources`.

**Reply by feniix** [reply]
so the values `XXXXXXXX` comes from the credential manager but `req` is something that we print out as a debug step

**Comment by aoldershaw** [comment]
Just to verify, have you enabled secret redaction (`CONCOURSE_ENABLE_REDACT_SECRETS=true`)?

**Reply by feniix** [reply]
yes that is correct, `CONCOURSE_ENABLE_REDACT_SECRETS` is set to true

**Comment by chenbh** [comment, marked as answer]
The secrets redaction logic only applies to stdout and stderr for [build steps (`get`, `put`, `task`)](https://github.com/concourse/concourse/blob/08aaab16f3dd171935f4c629aa5949a516224b9e/atc/engine/builder/delegate_factory.go#L476-L495), however resource `check` containers are handled differently and go through a whole separate codepath. From your description it sounds like it's printing out the creds in a resource `check` step, in which case there's no easy way to fix that other than to stop printing it.

Good news is that with the upcoming v7.0.0, we tore out the separate codepath and now run checks as builds (https://github.com/concourse/concourse/pull/6022). This should make it so that cred redaction now Just Works™️ for checks as well

**Reply by feniix** [reply]
thanks a lot! this is clarifying!
